### ISSUE TITLE
RegEx Syntax issue in roles/vdm/defaults/main.yaml

### ISSUE THREAD

**Issue by nachai9** [sassoftware/viya4-deployment#133]
I am running into same issue as issue # 87 when I am using Azure Container Registry. Can you pls reopen it for me.

Here is the sample error and attaching ansible-vars.yaml for reference.

ERROR! We were unable to read either as JSON nor YAML, these are the errors we got from each:
JSON: Expecting value: line 1 column 1 (char 0)

 

Syntax Error while loading YAML.
  found unknown escape character

 

The error appears to be in '/r/ge.unx.sas.com/vol/vol610/u61/nachai/viya4_3/viya4-deployment/roles/vdm/defaults/main.yaml': line 15, column 60, but may
be elsewhere in the file depending on the exact syntax problem.

 

The offending line appears to be:

 

V4_CFG_CR_URL: https://cr.sas.com
V4_CFG_CR_HOST: "{{ V4_CFG_CR_URL | regex_replace('^https?:\/\/(.*)\/?', '\\1') }}"
                                                           ^ here
We could be wrong, but this one looks like it might be an issue with
missing quotes. Always quote template expression brackets when they
start a value. For instance:

 

    with_items:
      - {{ foo }}

 

Should be written as:

 

    with_items:
      - "{{ foo }}"
[ansible-vars.zip](https://github.com/sassoftware/viya4-deployment/files/7076894/ansible-vars.zip)

**Comment by AWSmith0216** [comment]
Just a thought, but your V4_CFG_CR_URL value doesn't match the regex. Perhaps try adjusting your value to 'V4_CFG_CR_URL: https://nachai3acr.azurecr.io' and see what happens.

**Comment by nachai9** [comment]
I am getting following error with V4_CFG_CR_URL:

Monday 30 August 2021  10:08:13 -0400 (0:00:01.443)       0:00:21.601 *********
ERROR! We were unable to read either as JSON nor YAML, these are the errors we got from each:
JSON: Expecting value: line 1 column 1 (char 0)

Syntax Error while loading YAML.
  found unknown escape character

The error appears to be in '/r/ge.unx.sas.com/vol/vol610/u61/nachai/viya4_3/viya4-deployment/roles/vdm/defaults/main.yaml': line 15, column 60, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

V4_CFG_CR_URL: https://cr.sas.com
V4_CFG_CR_HOST: "{{ V4_CFG_CR_URL | regex_replace('^https?:\/\/(.*)\/?', '\\1') }}"
                                                           ^ here
We could be wrong, but this one looks like it might be an issue with
missing quotes. Always quote template expression brackets when they
start a value. For instance:

    with_items:
      - {{ foo }}

Should be written as:

    with_items:
      - "{{ foo }}"

**Comment by enderm** [comment]
I was able to replicate the issue running from cldlgn.fyi.sas.com.
I strongly suspect that it has something to do with the particular combination of dependencies - but that's a big ocean and a small fish.
I did come across this: https://github.com/ansible/ansible/issues/52868
It appears to describe exactly the same issue, and suggests using single quotes outside and double quotes inside. That worked for me. It is also confirmed by http://www.yamllint.com/ (the current syntax fails, swapping the quotes passes)
